### PR TITLE
Anti-Ghost-Ping Verbesserungen

### PR DESCRIPTION
**Pull request by martinbndr**
Rollen Whitelist - User, die Rollen besitzen die auf der Whitelist sind werden vom anti-ghost-ping ignoriert

### CODE REVIEW

**Comment by SCDerox** [comment]
Hi, thanks so much for your PR. The development language of this repo is English, that's while I am answering in another language. 
Before we can review your pull request, please make sure that all checks are passing. In this case the style-check failed ([read more](https://github.com/ScootKit/CustomDCBot/actions/runs/3509814649/jobs/5879170305)). You can always run the tests locally by executing `npm test`.

File: modules/anti-ghostping/events/messageDelete.js [hunk 1]
```diff
@@ -6,6 +6,12 @@ module.exports.run = async function (client, msg) {
     if (!messageWithMentions[msg.id]) return;
     const moduleStrings = client.configurations['anti-ghostping']['config'];
     if (messageWithMentions[msg.id].author.bot) return;
+	if (moduleStrings.ignoredChannels.includes(msg.channel.id)) return;
+    let whitelisted = false;
+    moduleStrings.ignoredRoles.forEach(r => {
+        if (msg.member.roles.cache.get(r)) whitelisted = true;
+    });
+    if (whitelisted) return;
```
Comment (SCDerox) [inline modules/anti-ghostping/events/messageDelete.js:14]: Code-Readability could be improved massively by using the `.find` / `.filter` function. 
Here's an example: 
```javascript
if (msg.member.roles.cache.find(r => moduleStrings.ignoredRoles.includes(r.id))) return;
```
Please note that I did not test this call and this is purely a suggestion. Feel free to pass feedback or state your opinion.

Comment (martinbndr) [reply modules/anti-ghostping/events/messageDelete.js]: Oh yea thats a lot better, I changed it now.

**Review by SCDerox — COMMENTED** [review]
Thanks so much for resolving these issues! We'll review and test (and hopefully merge) your pull request in the following days. 

*Please note that rollout on SCNX depends on our sole discretion and it may take up to two weeks to reach our main branch - beta branches usually receive your changes after merge).*
 
Because this is your first PR: If you want to receive a small bounty (= small amount of money) after we've merged your PR as part of our [Developer Programm](https://faq.scnx.app/open-source-developer-pool/), please [create an SCNX Marketplace Organization](https://scnx.app/de/user/organizations/), enable payouts and make sure your Discord-Account is linked to your GitHub.

File: modules/anti-ghostping/events/messageCreate.js [hunk 2]
```diff
@@ -5,6 +5,8 @@ module.exports.run = async function (client, msg) {
     if (msg.guild.id !== client.config.guildID) return;
     const moduleConfig = client.configurations['anti-ghostping']['config'];
     if (moduleConfig.ignoredChannels.includes(msg.channel.id)) return;
+    if (msg.member.roles.cache.find(r => moduleConfig.ignoredRoles.includes(r.id))) return;
+    if (moduleConfig.ignoredRoles.includes(msg.channel.id)) return;
```
Comment (SCDerox) [inline modules/anti-ghostping/events/messageCreate.js:9]: Why are we checking if the channel id is part of the ignored roles config setting 😅

File: modules/anti-ghostping/events/messageDelete.js [hunk 3]
```diff
@@ -6,6 +6,8 @@ module.exports.run = async function (client, msg) {
     if (!messageWithMentions[msg.id]) return;
     const moduleStrings = client.configurations['anti-ghostping']['config'];
     if (messageWithMentions[msg.id].author.bot) return;
+    if (moduleStrings.ignoredChannels.includes(msg.channel.id)) return;
+    if (msg.member.roles.cache.find(r => moduleStrings.ignoredRoles.includes(r.id))) return;
```
Comment (SCDerox) [inline modules/anti-ghostping/events/messageDelete.js:10]: I don't think these checks are needed here, as `messageWithMentions` only includes this message if checks got passed (see: `messageCreate.js`, line 7-10)

File: modules/anti-ghostping/config.json [hunk 4]
```diff
@@ -37,6 +37,17 @@
       },
       "type": "array",
       "content": "channelID"
+    },
+	{
```
Comment (SCDerox) [inline modules/anti-ghostping/config.json:41]: Formatting:
```suggestion
    {
```

**Review by SCDerox — CHANGES_REQUESTED** [review]
Thanks so much for your patience. After careful review of your code-changes I have found a few points I'd like to discuss. Please review my comments - feel free to discuss them if you do not agree with our assessments - I usually respond within a few hours. 
Please make sure to resolve every discussion before re-requesting a review.

**Comment by SCDerox** [comment]
Just a short check-in: There are multiple, unresolved conversations. Please either resolve them or comment on them to continue the review process. Once all conversations got resolved, I'll be able to merge this PR.

**Comment by SCDerox** [comment]
Closed as stale - please comment to reopen this PR.

**Pull request closed without merging**